### DISCUSSION TITLE
Alpine package

### DISCUSSION THREAD

**Discussion by juliekoubova** [psy0rz/zfs_autobackup#215] [Show and tell]
Hey, I just added zfs-autobackup to Alpine testing repo. 

https://pkgs.alpinelinux.org/package/edge/testing/x86_64/py3-zfs-autobackup

**Comment by psy0rz** [comment]
Awesome! Alpine happens to be my favorite distro to host my docker containers:

- Supports ZFS on root (Via the Alpine extended iso)
- Very up to date packages for both ZFS and docker. (and others)
- Rolling releases, so no reinstalls.
- Simple and clean boot + init system
- And ofcourse very small and efficient. (the speed of the apk package manager is amazing)

I use https://github.com/csachs/docker-zfs-plugin to have ZFS datasets for every ZFS volume.

So all my docker-compose stuff updates automatically every night, after snapshotting the volumes with ZFS. 

Whenever it fails i can simple rollback and fix whatever went wrong. :)

(I even run some legacy VM's via  qemu-kvm inside docker now)

**Comment by psy0rz** [comment]
And since the rootfs is also ZFS, i can also auto-upgrade Alpine (every week)  I created script that roll it back and reboot if an upgrade fails.

**Comment by psy0rz** [comment]
(Oh and obviously i use zfs-autobackup to backup all those docker volumes  )

**Comment by juliekoubova** [comment]
It's my favorite Linux as well, coming from FreeBSD 🙂 I also highly recommend https://docs.zfsbootmenu.org/en/v2.2.x/ , it's a tiny initramfs-based boot loader that allows you to select the ZFS dataset or snapshot to boot your actual Linux from. Kinda like ZFS support in the FreeBSD boot loader.

It's already saved my ass when an upgrade went wrong 🤣 And you don't need an ext4 boot partition, just the standard EFI one.

**Comment by psy0rz** [comment]
ah cool i wanted to try it, but wasnt sure if anyone had tried it with alpine :)

boot-environments are great

**Comment by juliekoubova** [comment]
They have a step-by-step guide for Alpine!On 13. 9. 2023, at 12:09 PM, DatuX ***@***.***> wrote:﻿
ah cool i wanted to try it, but wasnt sure if anyone had tried it with alpine :)
boot-environments are great

**Comment by psy0rz** [comment]
To follow up on this: All our boxes now use alpine, and we're publishing our installer script to make it easier to get Alpine + ZFS + ZfsBootmenu: https://github.com/psy0rz/alpinebox

Since installing is way too much steps otherwise, and for most VPS providers we just use images we curl > dd to disk. (will also be published later)
